quantities/quantities.hpp

```diff
@@ -142,21 +143,17 @@ template<>
 double FromM128D(__m128d x);
 template<>
 __m128d ToM128D(double x);
-
-// A type trait for testing if a type is a quantity.
-template<typename T>
-struct is_quantity : std::is_arithmetic<T>, not_constructible {};
-template<typename D>
-struct is_quantity<Quantity<D>> : std::true_type, not_constructible {};
+template<>
+__m128d ToM128D(int x);
```

Comment:
Declare two templatized overloads, one for arbitrary types (with an is_arithmetic `static_assert`) and one for quantities.

Since the `Quantity` one actually takes a `Quantity` (and not `Wide<Quantity>` or something like that when that appears, or at least will have a separate overload for that), there will be no need for SFINAE tricks here.

Reply:
Done.